File: Ordenamiento/Python/InsertionSort.py
# Insertion Sort
def insertionSort(arr):
	for i in range(1, len(arr)):
		print(i)
		j = i
		while j > 0 and arr[j-1] > arr[j]:
			arr[j], arr[j-1] = arr[j-1], arr[j]
			j -= 1
	return arr

arr = [64,34,1000,25,12,22,11,90]
print(insertionSort(arr))

File: Ordenamiento/Java/InsertionSort.java
public class InsertionSort{

    static void insertionSort(int arr[], int n) {
        for (int i = 1; i < n; i++){
            int j = i;
            while (j > 0 && arr[j-1] > arr[j]){
                int temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;

                j = j - 1;
            }
        }
    }

    // Print the array
    static void printArray(int arr[]) {
        System.out.println("Sorted array:");

        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String[] args) {
        int arr[] = {64, 34, 25, 12, 22, 11, 90};
        insertionSort(arr, arr.length);
        printArray(arr);  // Print the sorted array
    }
}
